### PR TITLE
Upgrade polkadot-v0.9.37

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #2435

TODO:
- [x] stable-asset: https://github.com/nutsfinance/stable-asset/pull/37
- [x] evm-tests: https://github.com/AcalaNetwork/evm-tests/pull/23
- [x] migrations: 
- https://github.com/paritytech/substrate/pull/12840
- https://github.com/paritytech/substrate/pull/12989
- [x] update try-runtime: https://github.com/paritytech/substrate/pull/12537
~~- [x] extrinsic ordering changes: The [PR](https://github.com/paritytech/substrate/pull/12451) removed `fill_block` from `frame-system`~~

### CODE REVIEW

File: runtime/acala/src/lib.rs [hunk 1]
```diff
@@ -129,7 +129,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	apis: RUNTIME_API_VERSIONS,
 	#[cfg(feature = "disable-runtime-api")]
 	apis: sp_version::create_apis_vec![[]],
-	transaction_version: 2,
+	transaction_version: 3,
```
Comment (xlc) [inline runtime/acala/src/lib.rs:132]: no need to bump this. no user should call system pallet (except remark). I don't want to break ledger just to ensure remark is working.

File: modules/currencies/src/tests.rs [hunk 2]
```diff
@@ -2575,3 +2575,8 @@ fn transfer_erc20_will_charge_gas() {
 		);
 	});
 }
+
+#[test]
+fn asset_exists_works() {
+	//TODO
```
Comment (xlc) [inline modules/currencies/src/tests.rs:2581]: TODO

File: runtime/acala/src/lib.rs (hunk 1 above)
Comment (xlc) [reply runtime/acala/src/lib.rs:132]: https://github.com/paritytech/substrate/pull/13207

Comment (zjb0807) [reply runtime/acala/src/lib.rs]: revert

**Comment by zjb0807** [comment]
acala try-runtime:
image

**Comment by zjb0807** [comment]
There is a new`ResetInactive` migration in polkadot-0.9.37, PR: https://github.com/paritytech/substrate/pull/13009.
We can do `pallet_balances::migration::MigrateToTrackInactive<Runtime, xcm_config::CheckingAccount>` in polkadot-0.9.37

**Review by xlc — APPROVED** [review]

**Pull request merged**